plugins/modules/web_infrastructure/jira.py

```diff
@@ -669,7 +669,7 @@ def request(
             try:
                 error = json.loads(info['body'])
             except Exception:
-                self.module.fail_json(msg=to_native(info['body']), exception=traceback.format_exc())
+                self.module.fail_json(msg=to_native(info.get('body')), exception=traceback.format_exc())
```

Comment:
How about instead checking `'body' in info` first (actually before this `try`) and producing some more helpful output, like which URL and method was used and what the status code was? This code here basically assumes that if the status is not 200, 201 or 204, it **must** be 400 or higher, which is probably violated in #3419. Without more information (like method, URL, status code) it's very hard to properly figure out what's the problem.

Reply:
Not sure if I follow what you meant by "probably violated in #3419". I agree we should add more information, though. What about:
```suggestion
                self.module.fail_json(msg=to_native(info), exception=traceback.format_exc())
```

Reply:
I meant that the issue in #3419 is likely that some other successful status code than 200, 201 or 204 was returned. `info['body']` is only around when the status code is >= 400.

Just dumping the whole `info` dictionary is potentially way too much information, but knowing the URL and the HTTP status, and if available the result, of the request that failed would be helpful.